packages/sdk/src/protocols/filter.ts

```diff
@@ -312,6 +359,55 @@ export class SubscriptionManager implements ISubscriptionSDK {
     clearInterval(this.keepAliveTimer);
     this.keepAliveTimer = null;
   }
+
+  private startMessageValidationCycle(): void {
+    const validationCycle = async (): Promise<void> => {
+      log.info("Starting message validation cycle");
+
+      for (const hash of this.receivedMessagesHashes.all) {
+        for (const [peerIdStr, hashes] of Object.entries(
+          this.receivedMessagesHashes.nodes
+        )) {
+          if (!hashes.has(hash)) {
+            log.error(
+              `Message with hash ${hash} not received from peer ${peerIdStr}.  Attempting renewal.`
+            );
+            const peerId = this.getPeers().find(
+              (p) => p.id.toString() === peerIdStr
+            )?.id;
+            if (!peerId) {
+              log.error(
+                `Unexpected Error: Peer ${peerIdStr} not found in connected peers.`
+              );
+              return;
+            }
+            try {
+              await this.renewPeer(peerId);
```

Comment:
it seems to be too aggressive to replace a peer due to missed message without any additional info

for example delay - what if message will come in but second later?

Reply:
this buffer improves but does not solve delay problem - it can occur that message arrives at 59th second of the time and another one at 60th

I would go with 3 missing messages or something